llvm/lib/Transforms/Scalar/LowerMatrixIntrinsics.cpp

```diff
@@ -493,10 +480,8 @@ class LowerMatrixIntrinsics {
   /// the result value of the instruction, with the only exceptions being store
   /// instructions and the matrix_column_major_store intrinsics. For those, the
   /// shape information indicates that those instructions should be lowered
-  /// using shape information as well.  A ValueMap is used so that when
-  /// sub-passes like optimizeTransposes performs RAUW the map stays
-  /// up-to-date.
-  ValueMap<Value *, ShapeInfo> ShapeMap;
+  /// using shape information as well.
```

Comment:
maybe say that special APIs should be used with RAUW in order to keep this consistent?

Reply:
Extended the comment, thanks!